docs/proposals/cluster-upgrades.md

```diff
@@ -0,0 +1,104 @@
+# Cluster Upgrades
+
+This is a position statement on Cluster Upgrades.  Submitted for feedback.
+
+## Assumptions
+
+Stable API types (v1, v2, etc) will to supported for fairly long times (e.g. 1 year, and probably longer).  This doc is not about that.
+
+Critical security patches will be backported to all recetnt minor releases (maybe 1 year?  TBD).  This doc is not about that.
+
+It is about "full upgrades" of the programs that make up the cluster (docker, kubelet, apiserver, controller-manager, addons, and their respective versions and flags and many options and input files).
+in order to get new features and performance improvements.
+
+User attitudes to upgrades differ.  Here are three user archetypes:
+  - "yearly upgrader": In my company, we only upgrade software once per year, because that is how we have always done it.  Unless there is a critical security fix.
```

Comment:
There's other reasons to be slow to upgrade (things are working fine now; we have deep acceptance process for new software; impacts lots of groups; etc)